db2/client.go

```diff
@@ -0,0 +1,253 @@
+package db2
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"math/big"
+	"net/http"
+
+	"github.com/ethereum/go-ethereum/common/hexutil"
+
+	"github.com/gobitfly/eth2-beaconchain-explorer/db2/store"
+)
+
+var ErrNotFoundInCache = fmt.Errorf("cannot find hash in cache")
+var ErrMethodNotSupported = fmt.Errorf("methode not supported")
+
+type RawStoreReader interface {
+	ReadBlockByNumber(chainID uint64, number int64) (*FullBlockRawData, error)
+	ReadBlockByHash(chainID uint64, hash string) (*FullBlockRawData, error)
+	ReadBlocksByNumber(chainID uint64, start, end int64) ([]*FullBlockRawData, error)
+}
+
+type WithFallback struct {
+	roundTripper http.RoundTripper
+	fallback     http.RoundTripper
+}
+
+func NewWithFallback(roundTripper http.RoundTripper, fallback http.RoundTripper) *WithFallback {
+	return &WithFallback{
+		roundTripper: roundTripper,
+		fallback:     fallback,
+	}
+}
+
+func (r WithFallback) RoundTrip(request *http.Request) (*http.Response, error) {
+	resp, err := r.roundTripper.RoundTrip(request)
+	if err == nil {
+		// no fallback needed
+		return resp, nil
+	}
+
+	var e1 *json.SyntaxError
+	if !errors.As(err, &e1) &&
```

Comment:
Should this condition use || instead of &&? Coz the current logic seems to require that all error checks must be true for the error to be handled

Reply:
We check if the error is a known error from our service. If it is, we can use fallback, otherwise we want to report the unexpected error.